sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -203,6 +210,8 @@ private[javasdk] final class ActionsImpl(
           case NonFatal(ex) =>
             // command handler threw an "unexpected" error
             Future.successful(handleUnexpectedException(service, in, ex))
+        } finally {
+          span.map(_.end())
```

Comment:
This is not where the call ends - `Future[ActionResponse]`, so should be once that future completes.

Also, note that now we are back into the question about passing trace context through thread local for non async and how to do it for async calls to propagate context to things the user call (other Kalix services for example).

Reply:
changed now